From 44cd3719eb94c8d5776b37c73993eea0617902f7 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Sat, 11 Apr 2020 23:21:42 +0200
Subject: [PATCH] frame-xchg.c: fix build with glibc < 2.24

SOL_NETLINK is used since commit
87a198111af1ea67053895f7435fb99e3cdd2159 resulting in the following
build failure with glibc < 2.24:

src/frame-xchg.c: In function 'frame_watch_group_io_read':
src/frame-xchg.c:328:27: error: 'SOL_NETLINK' undeclared (first use in this function)
   if (cmsg->cmsg_level != SOL_NETLINK)
                           ^

This failure is due to glibc that doesn't support SOL_NETLINK before
version 2.24 and
https://github.com/bminor/glibc/commit/f9b437d5efce93800b51ad2a437c8b1c9616bf80

Fixes:
 - http://autobuild.buildroot.org/results/3485088b84111c271bbcfaf025aa4103c6452072

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Upstream status:
https://lists.01.org/hyperkitty/list/iwd@lists.01.org/thread/ZRN25CA5I7PGUTFIRJLLXN7VHWGDAZDN]
---
 src/frame-xchg.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/frame-xchg.c b/src/frame-xchg.c
index 50351615..4c950fd3 100644
--- a/src/frame-xchg.c
+++ b/src/frame-xchg.c
@@ -42,6 +42,10 @@
 #include "src/netdev.h"
 #include "src/frame-xchg.h"
 
+#ifndef SOL_NETLINK
+#define SOL_NETLINK 270
+#endif
+
 struct watch_group {
 	/*
 	 * Group IDs, except 0, are per wdev for user's convenience.
-- 
2.25.1

